### PR TITLE
Checking out to specific commit (hash) works as it should.

### PR DESCRIPTION
**Pull request by pomek**
Fixes: #34.

### CODE REVIEW

File: lib/commands/update.js [hunk 1]
```diff
@@ -55,19 +55,28 @@ module.exports = {
 					if ( stdout ) {
 						throw new Error( `Package "${ data.packageName }" has uncommitted changes. Aborted.` );
 					}
-				} )
-				.then( () => {
+
```
Comment (Reinmar) [inline lib/commands/update.js]: I don't like merging these callbacks. The log handling is a separate task which, btw, should be done differently (related to https://github.com/cksource/mgit2/issues/13, but not exactly this).

Comment (pomek) [inline lib/commands/update.js]: Fixed.

**Comment by Reinmar** [comment]
First of all, there's an important bug that errors aren't passed correctly to the logger:

image

I fixed one issue like that yesterday, but here's another.

**Comment by Reinmar** [comment]
I'll work on this.

**Comment by Reinmar** [comment]
OK, turns out that the whole logging mechanism is broken. The issue is that the exec command logs the error and rejects the promise with a logger instance. That logger instance is then passed to the update command which don't expect it.

So, this issue must be solved in https://github.com/cksource/mgit2/issues/13. The caller must not need to care about logs and rejected promise should always be rejected with an error, not some custom object.

**Pull request merged**